tests/v2/validation/provisioning/rke1/custom_cluster_test.go

```diff
@@ -81,84 +72,43 @@ func (c *CustomClusterProvisioningTestSuite) TestProvisioningRKE1CustomCluster()
-		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRolesShared, c.standardUserClient, c.psact},
-		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRolesDedicated, c.client, c.psact},
-		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRolesDedicated, c.standardUserClient, c.psact},
+		{"1 Node all roles " + provisioning.AdminClientName.String(), nodeRolesAll, c.client},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), nodeRolesAll, c.standardUserClient},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), nodeRolesShared, c.client},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), nodeRolesShared, c.standardUserClient},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), nodeRolesDedicated, c.client},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), nodeRolesDedicated, c.standardUserClient},
 	}
-
-	var name string
 	for _, tt := range tests {
-		testSession := session.NewSession()
-		defer testSession.Cleanup()
-
-		client, err := tt.client.WithSession(testSession)
-		require.NoError(c.T(), err)
-
-		for _, nodeProviderName := range c.nodeProviders {
-			externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
-			providerName := " Node Provider: " + nodeProviderName
-			for _, kubeVersion := range c.kubernetesVersions {
-				for _, cni := range c.cnis {
-					name = tt.name + providerName + " Kubernetes version: " + kubeVersion + " cni: " + cni
-					c.Run(name, func() {
-						TestProvisioningRKE1CustomCluster(c.T(), client, externalNodeProvider, tt.nodeRoles, tt.psact, kubeVersion, cni, c.advancedOptions)
-					})
-				}
-			}
-		}
+		c.clustersConfig.NodesAndRolesRKE1 = tt.nodeRoles
+		permutations.RunTestPermutations(&c.Suite, tt.name, tt.client, c.clustersConfig, permutations.RKE1CustomCluster, nil, nil)
 	}
 }
 
 func (c *CustomClusterProvisioningTestSuite) TestProvisioningRKE1CustomClusterDynamicInput() {
-	require.GreaterOrEqual(c.T(), len(c.cnis), 1)
+	require.GreaterOrEqual(c.T(), len(c.clustersConfig.Basics.CNIs), 1)
 
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRolesRKE1
-
-	if len(nodesAndRoles) == 0 {
+	if len(c.clustersConfig.NodesAndRolesRKE1) == 0 {
 		c.T().Skip()
 	}
 
+	poolNames := permutations.GetRKE1TestName(c.clustersConfig.NodesAndRolesRKE1)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), c.client, c.psact},
-		{provisioning.StandardClientName.String(), c.standardUserClient, c.psact},
+		{provisioning.AdminClientName.String() + poolNames, c.client},
+		{provisioning.StandardClientName.String() + poolNames, c.standardUserClient},
```

Comment:
I still think we shouldn't add the pool names to the test name, but rather just log it. It will affect our regex a lot.